### PR TITLE
[22.05] libbpf 0.7.0 -> 0.8.1, bpftrace: 0.14.1 -> 0.15.0

### PR DESCRIPTION
**Pull request by dwf**
###### Description of changes

As per #175467. Useful as unbreaks 32-bit kernel builds (on e.g. ARM).

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/libbpf/default.nix [hunk 1]
```diff
@@ -1,6 +1,5 @@
 { fetchFromGitHub
-, fetchpatch
-, libelf
+, elfutils
```
Comment (martinetd) [inline pkgs/os-specific/linux/libbpf/default.nix:2]: I'm not sure what the policy is with backports: If you just want to unbreak 32 bit kernels this is the only line you need (-libelf +elfutils + same below in buildInputs)

That being said this version has been tested more and impact is minimal, so I'm happy either way

**Review by martinetd — APPROVED** [review]

**Comment by dwf** [comment]
Thanks for giving this a look! My understanding of the backports policy is that it has to be a commit that exists on `master` (so a bespoke commit with just those lines wouldn't be allowed). Sounds like we could get away with just the libbpf version bump commit, then, but I don't see anything wrong (or contradicting policy) with backporting the whole PR.

**Comment by vcunat** [comment]
Policy: the main point is to avoid breaking changes.  Otherwise it's basically at the discretion of the maintainers/committers.  https://nixos.org/manual/nixpkgs/stable/#submitting-changes-stable-release-branches

**Comment by vcunat** [comment]
I don't think libbpf really had breaking changes before the 1.0.0 jump.

**Comment by martinetd** [comment]
Yes, I also think libbpf 0.8 should be fine -- not sure what exactly changed with bpftrace but can't think of anything they'd have removed either.

**Comment by vcunat** [comment]
If we're updating to 0.8.0, I'd probably include 0.8.1: https://github.com/libbpf/libbpf/releases
(but no strong opinion)

**Comment by dwf** [comment]
@vcunat I've cherry-picked the libbpf 0.8.1 commit.

**Review by martinetd — APPROVED** [review]

**Comment by martinetd** [comment]
```
machine: must succeed: bpftrace -e 'kprobe:schedule {     printf("tgid: %d", ((struct task_struct*) curtask)->tgid); exit() }'
machine # stdin:1:42-64: ERROR: Unknown struct/union: 'struct task_struct'
machine # kprobe:schedule {     printf("tgid: %d", ((struct task_struct*) curtask)->tgid); exit() }
machine #                                          ~~~~~~~~~~~~~~~~~~~~~~
```

22.05 stable kernel does't have BTF? (didn't notice it wasn't done running when I reviewed 0.8.0)
That would also require to backport 47f9f047883996aa54defa9cda7ccd7e04df7901 if you care.

Well, I guess it's not a new problem with this PR so doesn't change my +1 :)

**Comment by dwf** [comment]
@martinetd It seems like if something's in a broken state we ought to fix it, so I've cherry-picked 47f9f047883996aa54defa9cda7ccd7e04df7901 too. Hopefully @vcunat concurs.

**Review by martinetd — APPROVED** [review]
Thanks! That made ofborg happy, no more remarks from me.

**Pull request merged**